### ISSUE TITLE
Application Redux state overwritten inside MetisMenu component

### ISSUE THREAD

**Issue by inigomarquinez** [alpertuna/react-metismenu#20]
I'm trying to create a React app using Redux state. In the app I want to use the MetisMenu component with a CustomLinkContainer, and be able to read and change the application state inside the MetisMenu elements.

But I have realized that inside the CustomLinkContainer the application state is not accessible.

Here is an example application. Inside the CustomLink the *appState* variable is undefined, instead of reading the current value from the state:

##### index.js
```javascript
let initialState = {
    appState: 'idle',
};

let store = createStore(stateReducer, initialState);

ReactDOM.render(
    <Provider store={ store }>
        <App />
    </Provider>,
  document.getElementById('root')
);
```

##### App.js
```javascript
const content = [
    {
        icon: 'icon-class-name',
        label: 'System',
        to: '#a-link',
    },
    {
        icon: 'icon-class-name',
        label: 'Tx / Rx',
        content: [
            {
                icon: 'icon-class-name',
                label: 'Magnet model',
                to: '#another-link',
            },
        ],
    },
];

const App = () => (
    <div>
        <MetisMenu content={content} LinkComponent={CustomLinkContainer} />
    </div>
);
```

##### CustomLinkContainer.js
```javascript
const mapStateToProps = (state) => {
    console.log('STATE INSIDE CUSTOM LINK CONTAINER');
    console.log(state);

    return {
        appState: state.appState
    }
}

const CustomLinkContainer = connect(
    mapStateToProps,
    null
)(CustomLink);
```
##### CustomLink.js
```javascript
class CustomLink extends React.Component {
    constructor(props) {
        super(props);
        this.onClick = this.onClick.bind(this);

        console.log('PROPS INSIDE CUSTOM LINK');
        console.log(this.props);
    }

    onClick(e) {
        if (this.props.hasSubMenu) {
            this.props.toggleSubMenu(e);
        } else {
            /*
             * your own operation using "to"
             * myGotoFunc(this.props.to);
             * or
             * this.props.activateMe(/ * Parameters to pass "onSelected" event listener * /);
             */

            this.props.activateMe({
                newLocation: this.props.to,
                selectedMenuLabel: this.props.label,
            });
        }
    }

    render() {
        return (
            <button className="metismenu-link" onClick={this.onClick}>
                {this.props.children}
            </button>
        );
    }
};
```
Is it possible to read / write the application Redux state inside the CustomLink component?

Thank you

**Comment by alpertuna** [comment]
Hi @inigomarquinez 
I have inspected your code and this is an issue you are right. Problem is that `react-metismenu` also uses `react-redux`. And at "STATE INSIDE CUSTOM LINK CONTAINER" point, the state is `react-metismenu`'s state since your `connect` catches `react-metismenu`'s `Provider`.

To solve this problem, I should put a property to ommit my `Provider`, and you should `combineReducers` `react-metismenu`'s and yours to provide just one store / Provider. If you are ok with this, I am able to add this feature. In this case I will be also looking for other solutions.

**Comment by inigomarquinez** [comment]
Hi @alpertuna 

Thanks for your quick response. If your suggestion works, it's ok for me if you add the feature.

I want to add more than one MetisMenu component in the application. Would it be a problem?

Thanks

**Comment by alpertuna** [comment]
You're welcome.
Hmm, ok I will add the feature thinking of multiple usage. In two days I will publish something does the job for you.

**Comment by alpertuna** [comment]
Hi @inigomarquinez 
Now external redux store is available with `v1.3.0-alpha.1`.

Usage example is (showing important points);
```javascript
import { createStore, combineReducers } from 'redux';
import MetisMenu from 'react-metismenu';
import metisMenuReducer from 'react-metismenu/lib/reducers';

const reducers = combineReducers({
    yourStore: yourReducers,
    // Your other reducer assignments...
    metisMenuStore: metisMenuReducer, // Here "metisMenuStore" is fixed and not customizable for now
});
const store = createStore(reducers, {
    yourStore: { // This name should be the same with above you assigned your reducer
        // Your application state
    },
   // Your other initial states...
   // There is no need to initalize "metisMenuStore"
});

<MetisMenu ... useExternalReduxStore={store} />
```

In this way, our stores will live in peace together :blush:

Nonetheless, it does not support multi usage for now because of redux action types are common and ambiguous. Fortunately, I will also add that support soon, usage may change a little.

I will be waiting for your feedback.

**Comment by inigomarquinez** [comment]
Hi @alpertuna ,

Thanks for the upgrade. I have tried it and it works good!

Please, let me know if you intend to do a new upgrade to support multiple `<MetisMenu>` components. It would be very useful.

Thank you,

**Comment by alpertuna** [comment]
Hi @inigomarquinez,
Finally, I was able to add multiple react-metismenu with redux support and published `v1.3.0-alpha.3`. Sorry for late return.

Readme parts you may be interested in;
- [Properties](https://github.com/alpertuna/react-metismenu/tree/1.3.0-alpha#properties)
- [Using With Redux](https://github.com/alpertuna/react-metismenu/tree/1.3.0-alpha#using-with-redux)

**Comment by inigomarquinez** [comment]
Thank you so much!
